### PR TITLE
sems-rhel7 env:  Add clang-10.0.0

### PR DESCRIPTION
**Pull request by jmgate**
## Motivation
From @rppawlo:
> Trilinos is looking to retire the clang 7 PR build. We are already running clang 10 in PR testing. I notice that the empire testing is using clang 7. Can Trilinos drop clang 7 testing? Should we wait until empire upgrades? What kind of schedule would that look like?

This PR updates the default clang in the `sems-rhel7` environment from `7.0.1` to `10.0.0`.

## Testing
Trying this out on an ascic machine, it seems to work just fine.

## Potential Concerns
* @e10harvey, I don't know if `sems-rhel7/all_supported_builds.sh` should be updated as well&mdash;I'm not sure what that file is used for.  If you want me to update it, let me know.
* Additionally, I wasn't sure if we should update the `cee-rhel7` environment as well, as the initial request was just for the sake of EMPIRE.  If you want me to update that env at the same time, let me know.
* BTW, I don't know how this might affect any of the ATDM nightly builds.

### CODE REVIEW

**Comment by jmgate** [comment]
CC:  @bathmatt, @dc-snl

**Comment by jmgate** [comment]
Note that this will turn on a number of new warnings as errors in the EMPIRE build.  That's fine&mdash;it'll just hold up our Trilinos sync until we get them taken care of.

**Review by e10harvey — CHANGES_REQUESTED** [review]
@jmgate, Thank you. In general, everything looks great. Since this is going into the officially supported builds (i.e. `cmake/std/atdm/sems-rhel7`), the only change I would suggest is updating the `sems-rhel7/all_supported_builds.sh` file to include:
```
sems-rhel7-clang-10.0.0-openmp-shared-release
sems-rhel7-clang-10.0.0-openmp-shared-release-debug
```

in `ATDM_CONFIG_ALL_SUPPORTED_BUILDS`. It may make sense to drop support for clang-7.0.1 at the same time.

Doing this will cause the nightlies to pick these changes up.

Once that's done, will you please test the changes on a sems-rhel7 machine via:
```bash
mkdir Trilinos-pr8775
cd Trilinos-pr8775
ln -s /path/to/Trilinos/cmake/std/atdm/ctest-s-local-test-driver.sh
nohup env ./ctest-s-local-test-driver.sh all &>ctest-s-local-test-driver.out &
```

and post the results or [cdash link](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&filtercount=2&showfilters=1&filtercombine=and&field1=groupname&compare1=63&value1=Experimental&field2=buildname&compare2=63&value2=sems-rhel7) here.

**Comment by jmgate** [comment]
Sounds good.  How long should `ctest-s-local-test-driver.sh all` take, roughly?

**Comment by e10harvey** [comment]
I believe ~6 hours

**Comment by jmgate** [comment]
Sorry for the delay here.  I launched the testing on an ASCIC machine, and have been unable to log back in since late Thursday.

**Comment by jmgate** [comment]
```
***
*** ./ctest-s-local-test-driver.sh
***

ATDM_TRILINOS_DIR = '/home/jmgate/Trilinos'

Load some env to get python, cmake, etc ...

Hostname 'ascic157' matches known ATDM host 'ascic157' and system 'sems-rhel7'
Setting compiler and build options for build-name 'default'
Using SEMS RHEL7 compiler stack GNU-7.2.0 to build DEBUG code with Kokkos node type SERIAL
NOTE: SEMS modules not defined, loading their definition!

Running builds:
    sems-rhel7-cuda-9.2-Volta70-complex-shared-release-debug
    sems-rhel7-gnu-7.2.0-openmp-complex-shared-release-debug
    sems-rhel7-intel-18.0.5-openmp-complex-shared-release-debug
    sems-rhel7-intel-18.0.5-openmp-shared-debug
    sems-rhel7-intel-18.0.5-openmp-shared-release-debug

Thu Feb 18 08:50:38 MST 2021

Running Jenkins driver Trilinos-atdm-sems-rhel7-cuda-9.2-Volta70-complex-shared-release-debug.sh ...

    Creating directory: Trilinos-atdm-sems-rhel7-cuda-9.2-Volta70-complex-shared-release-debug

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-sems-rhel7-cuda-9.2-Volta70-complex-shared-release-debug/smart-jenkins-driver.out

real    73m35.553s
user    949m49.680s
sys     95m23.264s

56% tests passed, 514 tests failed out of 1159

Thu Feb 18 10:04:14 MST 2021

Running Jenkins driver Trilinos-atdm-sems-rhel7-gnu-7.2.0-openmp-complex-shared-release-debug.sh ...

    Creating directory: Trilinos-atdm-sems-rhel7-gnu-7.2.0-openmp-complex-shared-release-debug

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-sems-rhel7-gnu-7.2.0-openmp-complex-shared-release-debug/smart-jenkins-driver.out

real    32m14.895s
user    407m10.665s
sys     35m42.164s

100% tests passed, 0 tests failed out of 1135

Thu Feb 18 10:36:29 MST 2021

Running Jenkins driver Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-complex-shared-release-debug.sh ...

    Creating directory: Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-complex-shared-release-debug

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-complex-shared-release-debug/smart-jenkins-driver.out

real    110m43.736s
user    1841m35.220s
sys     99m24.555s

100% tests passed, 0 tests failed out of 1135

Thu Feb 18 12:27:12 MST 2021

Running Jenkins driver Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-shared-debug.sh ...

    Creating directory: Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-shared-debug

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-shared-debug/smart-jenkins-driver.out

real    118m34.779s
user    1493m45.767s
sys     189m5.557s

100% tests passed, 0 tests failed out of 2138

Thu Feb 18 14:25:47 MST 2021

Running Jenkins driver Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-shared-release-debug.sh ...

    Creating directory: Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-shared-release-debug

    Creating directory: SRC_AND_BUILD

    See log file Trilinos-atdm-sems-rhel7-intel-18.0.5-openmp-shared-release-debug/smart-jenkins-driver.out

real    185m58.213s
user    3155m33.372s
sys     248m18.601s

32% tests passed, 1462 tests failed out of 2143

Thu Feb 18 17:31:45 MST 2021

Done running all of the builds!
```

I'm guessing the CUDA failures were due to running on a machine without CUDA cards.  I can't find the corresponding CDash link.

**Comment by e10harvey** [comment]
Thanks, @jmgate. Here is the cdash link: https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2021-02-18&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=sems-rhel7&field2=groupname&compare2=63&value2=Experimental. I think we can merge this and revert / follow-up later if we are seeing failures then. I will wait to approve until we have recent history of these builds on CDash.

**Comment by jmgate** [comment]
Do you need me to do anything more at this point?

**Comment by e10harvey** [comment]
No. Thanks again.

**Review by e10harvey — APPROVED** [review]

**Pull request merged**
